### PR TITLE
uniswap v3 resolver updates

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- added new functions

### CODE REVIEW

**Comment by pradyuman-verma** [comment]
Deployed on polygon: 
Addr: [0x42D65eD5b96A990CaCe1CF0ad936f324709d30B4](https://polygonscan.com/address/0x42D65eD5b96A990CaCe1CF0ad936f324709d30B4#code)

Deployed on Arb: 
Addr: [0xCf8A882763C92857FEEbd54620e2c09D8E12E60B](https://arbiscan.io/address/0xCf8A882763C92857FEEbd54620e2c09D8E12E60B#code)

Deployed on Opt: 
Addr: [0xbD07728E20c49F0Fa22c82915955fbeA5E203a6a](https://optimistic.etherscan.io/address/0xbD07728E20c49F0Fa22c82915955fbeA5E203a6a#code)

Deployed on Mainnet: 
Addr: [0x2B1c53652F485e9CAF40888b39E1b81e17C7b785](https://etherscan.io/address/0x2B1c53652F485e9CAF40888b39E1b81e17C7b785)

File: contracts/protocols/polygon/uniswap/helpers.sol [hunk 1]
```diff
@@ -248,26 +248,52 @@ abstract contract Helpers is DSMath {
...
+                    SingleAmountParams(
+                        amount1,
+                        mintParams.slippage,
+                        false,
+                        TickMath.getSqrtRatioAtTick(newCurrentTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.lowerTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.upperTick)
+                    )
+                );
+            }
```
Comment (thrilok209) [inline contracts/protocols/polygon/uniswap/helpers.sol]: @pradyuman-verma You can do it simply way, right? Why code duplicate? Or am I missing something?

Comment (pradyuman-verma) [reply contracts/protocols/polygon/uniswap/helpers.sol]: In one case, the reverseFlag is false. So have to do it twice.

File: contracts/protocols/polygon/uniswap/helpers.sol [hunk 2]
```diff
@@ -231,6 +231,73 @@ abstract contract Helpers is DSMath {
...
+                ? (mintParams.tokenA, mintParams.tokenB)
+                : (mintParams.tokenB, mintParams.tokenA);
+
+            if (mintParams.tokenA < mintParams.tokenB) {
+                amount0 = mintParams.amountA;
+                (amount1, liquidity) = calculateSingleAmount(
+                    SingleAmountParams(
+                        amount0,
+                        mintParams.slippage,
+                        true,
```
Comment (thrilok209) [inline contracts/protocols/polygon/uniswap/helpers.sol]: @pradyuman-verma Here it should be `false`, right?

**Pull request merged**